src/Event/Dispatcher.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace PhpAb\Event;
+
+class Dispatcher implements DispatcherInterface
+{
+    /**
+     * @var array An array holding the Listeners for a given event.
+     *            The format looks like
+     *            [
+     *              'eventname' => [callable, callable, callable],
+     *              'eventname2' => [callable, callable, callable]
+     *            ]
+     */
+    private $listeners = [];
+
+    /**
+     * @param          $eventName The name of the event to listen for
+     * @param callable $callable  The Callable to execute once the event takes place
+     */
+    public function addListener($eventName, callable $callable)
```

Comment:
We need a priority, else you cannot control if your listener should be executed before or after an event.